### PR TITLE
QA-235: pipeline: Switch to static checks by golangci-lint

### PR DESCRIPTION
**Pull request by lluiscampos**
And fix the code accordingly.

### CODE REVIEW

File: utils/restutil/view/view.go [hunk 1]
```diff
@@ -41,20 +41,34 @@ type RESTView struct {
 }
 
 func (p *RESTView) RenderSuccessPost(w rest.ResponseWriter, r *rest.Request, id string) {
-	w.Header().Add(HttpHeaderLocation, fmt.Sprintf("./%s/%s", strings.TrimLeft(r.URL.Path, "/api/0.0.1/"), id))
+	w.Header().Add(
+		HttpHeaderLocation,
+		fmt.Sprintf("./%s/%s", strings.TrimSuffix(r.URL.Path, "/api/0.0.1/"), id),
+	)
```
Comment (lluiscampos) [inline utils/restutil/view/view.go:47]: Dear reviewer, pay attention to this: I replaced `TrimLeft` with `TrimSuffix` :thinking:

File: utils/restutil/view/view.go [hunk 2]
```diff
@@ -41,20 +41,34 @@ type RESTView struct {
 }
 
 func (p *RESTView) RenderSuccessPost(w rest.ResponseWriter, r *rest.Request, id string) {
-	w.Header().Add(HttpHeaderLocation, fmt.Sprintf("./%s/%s", strings.TrimLeft(r.URL.Path, "/api/0.0.1/"), id))
+	w.Header().Add(
+		HttpHeaderLocation,
+		fmt.Sprintf(".%s/%s", strings.TrimSuffix(r.URL.Path, "/api/0.0.1/"), id),
```
Comment (lluiscampos) [inline utils/restutil/view/view.go:46]: Dear reviewer, pay attention to this: I replaced `TrimLeft` with `TrimSuffix`, and modified slightly the format. Is this correct? :thinking:

Comment (lluiscampos) [reply utils/restutil/view/view.go:46]: It passes its unit test :+1:

File: store/mongo/migration_1_2_3.go [hunk 3]
```diff
@@ -61,7 +62,7 @@ func (m *migration_1_2_3) Up(from migrate.Version) error {
 		return err
 	}
 	var artifacts []*model.Image
-	if cursor.All(ctx, &artifacts); err != nil {
+	if err = cursor.All(ctx, &artifacts); err != nil {
```
Comment (tranchitella) [inline store/mongo/migration_1_2_3.go:65]: This was a bug, well spotted!

File: utils/restutil/view/view.go [hunk 4]
```diff
@@ -41,20 +41,34 @@ type RESTView struct {
 }
 
 func (p *RESTView) RenderSuccessPost(w rest.ResponseWriter, r *rest.Request, id string) {
-	w.Header().Add(HttpHeaderLocation, fmt.Sprintf("./%s/%s", strings.TrimLeft(r.URL.Path, "/api/0.0.1/"), id))
+	w.Header().Add(
+		HttpHeaderLocation,
+		fmt.Sprintf("./%s/%s", strings.TrimLeft(r.URL.Path, "/api0.1"), id),
+	)
```
Comment (tranchitella) [inline utils/restutil/view/view.go:47]: This TrimLeft looks wrong (and it already was before your change).
```suggestion
	w.Header().Add(
		HttpHeaderLocation,
		fmt.Sprintf(".%s/%s", strings.TrimPrefix(r.URL.Path, "/api/0.0.1"), id),
	)
```

**Review by tranchitella — APPROVED** [review]
Approved, with one comment/improvement.

File: utils/restutil/view/view.go (hunk 4 above)
Comment (lluiscampos) [reply utils/restutil/view/view.go:47]: If you see the force-pushed in the PR, I tried first using `TrimPrefix` but several unit tests failed. I tried with slight variations of adding a `/` in the format or in the prefix, but couldn't get all tests happy.

Can you try on your end? Maybe is just the tests that have the wrong expectations?

**Comment by lluiscampos** [comment]
@mender-test-bot start pipeline :five:

File: utils/restutil/view/view.go (hunk 4 above)
Comment (lluiscampos) [reply utils/restutil/view/view.go]: Fixed using `fmt.Sprintf(".%s/%s", strings.TrimPrefix(strings.TrimPrefix(r.URL.Path, "."), "/api"), id)`

**Comment by lluiscampos** [comment]
@mender-test-bot start pipeline :five: :repeat:

**Comment by lluiscampos** [comment]
@mender-test-bot start pipeline :five: :repeat:

**Pull request merged**